File: testing/unit-test-patterns/c#/UnitTestPatterns/Fake/FakeImplementation/FakeUniqueCloneValidationRulesRepository.cs
using System.Collections.Generic;
using UnitTestPatterns.Common.Interfaces;
using UnitTestPatterns.Common.Models;

namespace UnitTestPatterns.Fake.FakeImplementation
{
    public class FakeUniqueCloneValidationRulesRepository : IUniqueCloneValidationRulesRepository
    {
        private readonly string[] _validMaleFirstNames = {"John", "Jimmy", "Albert", "Tony"};
        private readonly string[] _validFemaleFirstNames = {"Jane", "Rebecca", "Ashley", "Emma"};
        private readonly string[] _validLastNames = {"Smith", "Anderson", "Doe", "Jones"};
        private const int MaxNumberOfEyes = 2;

        public IEnumerable<string> GetValidFirstNames(Gender gender)
        {
            return gender == Gender.Female ? _validFemaleFirstNames : _validMaleFirstNames;
        }

        public IEnumerable<string> GetValidLastNames()
        {
            return _validLastNames;
        }

        public int GetMaximumNumberOfEyes()
        {
            return MaxNumberOfEyes;
        }
    }
}

File: testing/unit-test-patterns/javascript/src/fakes/fake-implementation/FakeUniqueCloneValidationRulesRepository.ts
import { IUniqueCloneValidationRulesRepository } from '../../common/interfaces/UniqueCloneValidationRulesRepository.interface'
import { Gender } from '../../common/models/gender'

export class FakeUniqueCloneValidationRulesRepository
  implements IUniqueCloneValidationRulesRepository
{
  private readonly _validMaleFirstNames = ['John', 'Jimmy', 'Albert', 'Tony']
  private readonly _validFemaleFirstNames = [
    'Jane',
    'Rebecca',
    'Ashley',
    'Emma',
  ]
  private readonly _validLastFirstNames = ['Smith', 'Anderson', 'Doe', 'Jones']
  private readonly _maxNumberOfEyes = 2

  getValidFirstNames(gender: Gender) {
    return gender === Gender.male
      ? this._validMaleFirstNames
      : this._validFemaleFirstNames
  }

  getValidLastNames() {
    return this._validLastFirstNames
  }

  getMaximumNumberOfEyes() {
    return Promise.resolve(this._maxNumberOfEyes)
  }
}
